services/inc/src/main/java/org/cryptimeleon/incentive/services/promotion/PromotionService.java

```diff
@@ -164,7 +166,13 @@ private SideEffect handleSpendRequest(BigInteger promotionId, UUID basketId, UUI
         // using tid as user choice TODO change this once user choice generation is properly implemented, see issue 75
         DeductOutput spendProviderOutput = incentiveSystem.generateSpendRequestResponse(promotion.getPromotionParameters(), spendRequest, new ProviderKeyPair(providerSecretKey, providerPublicKey), tid, spendDeductTree, tid);
 
-        // TODO process provider output
+        // send transaction data to double-spending protection service
+        dsProtectionClient.dbSync(
+                tid,
+                spendRequest.getDsid(),
+                spendProviderOutput.getDstag(),
+                tid.toString() // TODO change this once user choice generation is properly implemented
+        );
```

Comment:
I think we should make this call asynchronous :)

Reply:
And I am thinking about how to test this sufficiently, maybe with a mock DsProtectionClient?